zonal_datacube/zonal_datacube.py

```diff
@@ -0,0 +1,257 @@
+                geom_masked_datacube = ZonalDataCube._mask_datacube_by_geom(
+                    zone_geometry, masked_datacube, tile.bounds
+                )
+
+                zone_attributes = zone.drop("zone_wkt")
+                result = zone_attributes.copy()
+                for func in funcs:
+                    func_result = func.func(zone_attributes, geom_masked_datacube)
+                    result = pd.concat([result, func_result])
+
+                partition_results.append(result)
+
+        return pd.DataFrame(partition_results)
+
+    @staticmethod
+    def _get_dask_zones(zones, bounds, cell_size, npartitions):
+        # fishnet features to make them partition more efficiently in Dask
+        fishnetted_zones = fishnet(zones, *bounds, cell_size)
+
+        # Dask can't understand the geometry type,
+        # so just serialize to WKT and drop geometry
+        fishnetted_zones["zone_wkt"] = fishnetted_zones["geometry"].apply(
+            lambda x: x.wkt
+        )
+        del fishnetted_zones["geometry"]
+
+        dask_zones = dd.from_pandas(fishnetted_zones, npartitions=npartitions)
+
+        # Index based on the fishnet geometry so features
+        # that overlap the same fishnet grid cell
+        # are processed in the same partition.
+        # This ensures we only need to read the raster data
+        # only once per grid cell across the cluster
+        indexed_zones = dask_zones.set_index("fishnet_wkt", npartitions=npartitions)
+
+        return indexed_zones
+
+    @staticmethod
+    def _get_meta(zones, funcs):
+        meta = dd.utils.make_meta(zones)
+
+        # drop geom columns for meta
+        if "zone_wkt" in meta:
+            meta = meta.drop("zone_wkt", axis=1)
+        elif "fishnet_wkt" in meta:
+            meta = meta.drop("fishnet_wkt", axis=1)
+
+        for func in funcs:
+            if func.meta:
+                for col, type in func.meta.items():
+                    meta[col] = pd.Series(dtype=type)
+            else:
+                # TODO how do we determine dtype if they didn't specify meta?
+                meta[col] = pd.Series(dtype="float64")
+
+        return meta
+
+
+    def _get_stac_items(self, stac_items, spatial_only):
+        """Create copies of the STAC items and mutate any properties for our
+        processing."""
+        new_items = []
+
+        for original_item in stac_items:
+            new_item = original_item.full_copy()
+
+            # if spatial only, set all items to the same datetime so STAC clients
+            # will see them as aligned on the same time dimension
+            if spatial_only:
+                new_item.datetime = self.DEFAULT_TIME
+
+            new_items.append(new_item)
+
+        return new_items
+
+    @staticmethod
+    def _mask_datacube_by_geom(geom, datacube, bounds, all_touched=False):
+        if not geom.is_valid:
+            geom = geom.buffer(0)
+
+        height = datacube.longitude.shape[0]
```

Comment:
the dimensions look swapped to me. shouldn't height be the number of rows or datacube.latitude.shape[0]?